index.html

```diff
@@ -2543,18 +2588,271 @@ <h1>
     </h1>
 
     <p>
-A <a>DID resolver</a> is a software or hardware component with an API for
-resolving <a>DIDs</a> of at least one <a>DID method</a>. It executes the read
-operation for the <a>DID method</a> corresponding to the <a>DID</a> being
-resolved to obtain the authoritative <a>DID document</a>. For more information,
-see Section <a href="#read-verify"></a>.
+This section defines the input and output contracts of the 
+<a>DID resolution</a> and <a>DID URL dereferencing</a> functions. The
+exact implementation of these functions is out of scope for this specification,
+but some considerations for implementors are available in [[?DID-RESOLUTION]].
     </p>
-
+    
     <p>
-The interfaces and algorithms for resolving <a>DIDs</a> and dereferencing
-<a>DID URLs</a> are specified in [[?DID-RESOLUTION]].
+All conformant <a>DID resolvers</a> MUST implement the <a>DID resolution</a> function
+for at least one <a>DID method</a> and be able to return a <a>DID document</a> in 
+at least one conformant representation. All conformant <a>DID URL dereferencers</a>
+MUST implement dereferencing for at least one conformant representation type.
     </p>
 
+    <section>
+        <h2>
+DID Resolution
+        </h2>
+        
+        <p>
+The <a>DID resolution</a> function resolves a <a>DID</a> into a <a>DID document</a>
+by using the "Read" operation of the applicable <a>DID method</a>. (See <a href="#read-verify"></a>.)
+The details of how this process is accomplished is outside the scope of this
+specification, but all implementations MUST implement a function in the form:
+        </p>
+        
+        <p>
+<code>resolve ( did, input-options ) -> ( did-document, did-document-metadata, did-resolution-metadata )</code>
+        </p>
+        
+        <p>
+The inputs of this function are a <a>DID</a> and a set of input options. The 
+<a>DID</a> is REQUIRED. Note that if a client wishes to resolve a <a>DID URL</a>, 
+the client MUST first transform the <a>DID URL</a> into a bare <a>DID</a>, 
+including removal of any fragment component. The input options
+are a map of key-value string pairs as described in <a href="#metadata-structure"></a>.
+The input options are REQUIRED but the map MAY be empty. Concrete <a>bindings</a> MUST NOT
+define additional inputs to this function.
+        </p>
+        
```

Comment:
Given my strong opinion on the subject, I'll plan to attend.

Reply:
I strongly disagree with @msporny here. The language here defines things in terms of a functional API and not a network protocol. As discussed many times this falls under this bullet of our charter:

*  Establish a deterministic mapping between DID method identifiers and the resolution process used to resolve that DID method. 

It can't be deterministic without normative language. The suggestion to just point to a NOTE and "Signal" to people that it's "going to be a real spec some day" is just acrobatics and avoiding the issue, which is not what we had agreed upon in Amsterdam.

Reply:
The language in this section seems abstract enough to define what is required for DID Resolution without defining a protocol (client-server) although that can certainly be interpreted or derived. It seems like some language like this defines what is meant/required for DID Resolution, and then a NOTE can be used to be more specific in what a DID Resolution protocol may be. My $0.02.